File: exercise2/post_processing/python/plotexample.py
import matplotlib.pyplot as plt
from chem1d_utilities import *

# Read chem1d data
yiend = readchem1d("../exercise1/yiend.dat")

# Put spatial coordinate in array x
x = yiend.var('x(i)')

print('Flame temperature: {:.6e} K'.format(yiend.var('Temp')[-1][0]))
print('Burning velocity : {:.6e} cm/s'.format(yiend.var('Massflow')[0][0] / yiend.var('Density')[0][0]))
print('CO emission      : {:.6e}'.format(yiend.var('CO')[-1][0]))
print('NO emission      : {:.6e}'.format(yiend.var('NO')[-1][0]))

# Plot T vs x
plt.figure(1)
plt.plot(x, yiend.var('Temp'), '.-', label='Temperature')
plt.xlabel('x [cm]')
plt.ylabel('T [K]')
plt.title('Temperature vs x')
plt.grid(True)

# Plot species mass fractions vs x
species_names = ['CH4', 'O2', 'CO2', 'H2O', 'CO', 'OH', 'NO']

plt.figure(2)
for species_name in species_names:
    plt.plot(x, yiend.var(species_name), '.-', label=species_name)  # Adjust indexing if needed
plt.xlabel('x [cm]')
plt.ylabel('Mass fraction [-]')
plt.title('Species Mass Fractions vs x')
plt.legend()
plt.grid(True)

# Uncomment for log scale:
# plt.yscale('log')
# plt.ylim([1e-8, 1])

# Plot NO mass fraction vs x
plt.figure(3)
plt.plot(x,yiend.var('NO'), '.-', label='NO')
plt.xlabel('x [cm]')
plt.ylabel('NO mass fraction [-]')
plt.title('NO Mass Fraction vs x')
plt.grid(True)

plt.show()

File: exercise2/post_processing/matlab/plotexample.m
%% Load chem1d output data in matrix y
% Variable names are loaded into cell array a
[y,t,a] = readchem1d('yiend.dat');

% Assign some pointers
iTemp = find(strcmpi('temp',a));
iDensity = find(strcmpi('density',a));
iMassFlow = find(strcmpi('massflow',a));
iEnthalpy = find(strcmpi('Enthalpy',a));
iCH4  = find(strcmpi('CH4',a));
iO2   = find(strcmpi('O2',a));
iCO2  = find(strcmpi('CO2',a));
iH2O  = find(strcmpi('H2O',a));
iCO   = find(strcmpi('CO',a));
iNO   = find(strcmpi('NO',a));
iOH   = find(strcmpi('OH',a));
% Add more species here, if you want

% Put spatial coordinate in array x
x = y(:,strcmpi('x(i)',a));
% Put mixture fraction in array Z
Z = y(:,strcmpi('MixFrac',a));

%% Plot T vs x
figure(1);
plot(x, y(:,iTemp), '.-');
xlabel('x [cm]');
ylabel('T [K]');

%% Plot species mass fractions vs x
figure(2);
% Select some species
ivar = [iCH4, iO2, iCO2, iH2O, iCO, iOH];
plot(x, y(:,ivar), '.-');
xlabel('x [cm]');
ylabel('Mass fraction [-]');
% Add a legend
legend(a{ivar});

% Uncomment next line for log scale
% set(gca, 'Yscale', 'log', 'Ylim', [1e-8 1]);

%% Plot NO mass fraction vs x
figure(3);
plot(x, y(:,iNO), '.-');
xlabel('x [cm]');
ylabel('NO mass fraction [-]');


